Source/VirtualTrees.WorkerThread.pas

```diff
@@ -1,225 +1,231 @@
-unit VirtualTrees.WorkerThread;
-
-interface
-
-uses
-  System.Classes,
-  VirtualTrees;
-
-type
-  // internal worker thread
-  TWorkerThread = class(TThread)
-  private
-    FCurrentTree: TBaseVirtualTree;
-    FWaiterList: TThreadList;
-    FRefCount: Integer;
-    class procedure EnsureCreated();
-    class procedure Dispose();
-    procedure WaitForValidationTermination(Tree: TBaseVirtualTree);
-  protected
-    procedure Execute; override;
-  public
-    constructor Create();
-    destructor Destroy; override;
-
-    /// For lifeteime management of the TWorkerThread
-    class procedure AddThreadReference;
-    class procedure ReleaseThreadReference();
-
-    class procedure AddTree(Tree: TBaseVirtualTree);
-    class procedure RemoveTree(Tree: TBaseVirtualTree);
-
-    property CurrentTree: TBaseVirtualTree read FCurrentTree;
-  end;
-
-
-
-
-
-implementation
-
-uses
-  Winapi.Windows,
-  System.Types,
-  System.SysUtils;
-
-type
-  TBaseVirtualTreeCracker = class(TBaseVirtualTree)
-  end;
-
-var
-  WorkerThread: TWorkerThread = nil;
-  WorkEvent: THandle;
-//----------------- TWorkerThread --------------------------------------------------------------------------------------
-
-class procedure TWorkerThread.EnsureCreated();
+unit VirtualTrees.WorkerThread;
+
+interface
+
+uses
+  System.Classes,
+  VirtualTrees;
+
+type
+  // internal worker thread
+  TWorkerThread = class(TThread)
+  private
+    FCurrentTree: TBaseVirtualTree;
+    FWaiterList: TThreadList;
+    FRefCount: Integer;
+    class procedure EnsureCreated();
+    class procedure Dispose(CanBlock: Boolean);

```

Comment:
You're prefixing the parameter in the other procedure (``ReleaseThreadReference``) with an ``A`` why don't you do this here too? Also this could be a ``const`` parameter.

Reply:
The inconsistency is my bad.

Regarding const: I use it only when it makes some technical sense. Types > SizeOf(Pointer) or if the type is managed. I know that in theory this allows for changes of CanBlock inside the class medthod. But that has no effect on the callerside.